packages/react/src/auth/useSignOut.ts

```diff
@@ -0,0 +1,39 @@
+import { useCallback, useContext, useEffect } from "react";
+import { GadgetConfigurationContext, useApi } from "../../src/GadgetProvider";
+import { useAction } from "../../src/useAction";
+import { useUser } from "./useUser";
+
+/**
+ * Returns a callback that will call the configured `signOutActionApiIdentifier` on the `User` model and optionally redirect (by default).
+ * @returns The `Promise` to call to execute the `signOut` action. Will automatically redirect to the configured `signInPath` unless the `redirect: false` option is provided.
```

Comment:
maybe a bit misleading as this describes the return value of the function that's returned